### PR TITLE
Add tmate & gnome updates

### PR DESCRIPTION
**Pull request by satmandu**
- Adds new `tmate` package, which is a variant of tmux with a remote shared screen.
image

- Updates `atkmm`, `cairomm_1_0`, `cairomm_1_16`, `glibmm_2_68`, `gstreamer`, `gtk4`, `gtkmm4`, `libevent`, `mm_common`, `msgpack_c`, `pangomm_1_4`, `pangomm_2_48`, and `transmission`

Builds properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=tmate crew update
```

### CODE REVIEW

**Comment by Zopolis4** [comment]
What is the purpose of splitting `cairomm` into multiple versions? Is it still needed?

File: packages/msgpack_c.rb [hunk 1]
```diff
@@ -1,50 +1,38 @@
...
 
-  def self.build
-    FileUtils.mkdir('builddir')
-    Dir.chdir('builddir') do
-      system "cmake #{CREW_CMAKE_OPTIONS} \
-      -DCMAKE_C_STANDARD='17' \
+  cmake_options "-DCMAKE_C_STANDARD='17' \
       -DBUILD_SHARED_LIBS=ON \
+      -DMSGPACK_32BIT=#{ARCH == 'i686' ? 'ON' : 'OFF'} \
       -DMSGPACK_ENABLE_STATIC=OFF \
```
Comment (Zopolis4) [inline packages/msgpack_c.rb:35]: Already disabled by our earlier passing of `-DBUILD_SHARED_LIBS=ON`.
https://github.com/msgpack/msgpack-c/blob/c-6.0.0/CMakeLists.txt#L90-L93

File: packages/msgpack_c.rb [hunk 2]
```diff
@@ -1,50 +1,38 @@
...
     system "sed -i 's,CMAKE_INSTALL_PREFIX}/lib,CMAKE_INSTALL_PREFIX}/#{ARCH_LIB},g' CMakeLists.txt"
     system "sed -i 's,CMAKE_INSTALL_LIBDIR lib,CMAKE_INSTALL_LIBDIR #{ARCH_LIB},g' CMakeLists.txt"
   end
 
-  def self.build
-    FileUtils.mkdir('builddir')
-    Dir.chdir('builddir') do
-      system "cmake #{CREW_CMAKE_OPTIONS} \
-      -DCMAKE_C_STANDARD='17' \
+  cmake_options "-DCMAKE_C_STANDARD='17' \
```
Comment (Zopolis4) [inline packages/msgpack_c.rb:32]: Any reason for doing this?

File: packages/msgpack_c.rb [hunk 3]
```diff
@@ -1,50 +1,38 @@
...
+       i686: '23f44da7a26b62bc6b38be11b7c2a5cefc1b2ac11a5d87982f73a4a2edb8ad03',
+     x86_64: '4171f5ad1d9e9b4e40b06e0e9f68173edace4f25d7d95e8b1f9e581e45a87f7d'
   })
 
-  depends_on 'cmake' => :build
   depends_on 'doxygen' => :build
-  depends_on 'graphviz' => :build
+  depends_on 'glibc' # R
 
   def self.patch
```
Comment (Zopolis4) [inline packages/msgpack_c.rb:26]: Send these improvements upstream? (i.e. add support for passing `-DCMAKE_INSTALL_PREFIX` the standard way)

File: packages/tmate.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on 'msgpack_c' # R
+  depends_on 'ncurses' # R
+
+  def self.patch
+    # Patch for msgpack_c 6.x
+    downloader 'https://github.com/tmate-io/tmate/pull/281.patch', '9f4f315a92b959606d42d672cf8c7312afaf4a93fba6c0e21bfee0b84e2d4667'
+    system 'patch -p1 -i 281.patch'
+  end
+
+  pre_configure_options "CFLAGS='-I#{CREW_PREFIX}/include/ncurses' CPPFLAGS='-I#{CREW_PREFIX}/include/ncurses'"
```
Comment (Zopolis4) [inline packages/tmate.rb:36]: It's [this issue](https://gitlab.kitware.com/cmake/cmake/-/issues/23236) again...

**Review by Zopolis4 — CHANGES_REQUESTED** [review]
Also, why are we disabling building documentation? That's not what we normally do...


Same thing with tests-- if it doesnt result in additional installed binaries or anything, unlike examples, we should be building and running them.

**Comment by satmandu** [comment]
Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.

**Comment by satmandu** [comment]
> What is the purpose of splitting `cairomm` into multiple versions? Is it still needed?

There are different version libraries.... With packages dependent upon particular versions...

File: packages/msgpack_c.rb (hunk 2 above)
Comment (satmandu) [reply packages/msgpack_c.rb:32]: Didn't change the existing option...

File: packages/msgpack_c.rb (hunk 3 above)
Comment (satmandu) [reply packages/msgpack_c.rb:26]: Feel free!

**Comment by Zopolis4** [comment]
> Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.

We install documentation for other packages, and we compress it to save size...

Let's keep the status quo for these packages for now, though.

**Comment by Zopolis4** [comment]
What about building tests?

**Comment by satmandu** [comment]
> > Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.
> 
> We install documentation for other packages, and we compress it to save size...
> 
> Let's keep the status quo for these packages for now, though.

Respectfully, I'd like to move in the other direction. We should set the `gnome` flagged packages to automatically stop building documentation. NOBODY uses the enclosed documentation for gnome libraries. Everyone just googles for documentation anyways.

**Comment by satmandu** [comment]
> What about building tests?

I'm honestly not sure the tests in these libraries are meaningful. They're not command line apps. Are we really going to follow up upstream with failed tests on widely used and tested gnome packages?

I'd be perfectly happy having tests run by default, and then set the gnome packages to not run tests by default. I just think it's a waste of CPU time...

**Comment by Zopolis4** [comment]
We run tests for packages for the purpose of ensuring we haven't broken anything-- a package can build fine but be broken at runtime, and tests help to catch some of that.

**Comment by satmandu** [comment]
I have removed _the don't build tests_ lines from the files in this PR, so the next time these are built, they can be tested. You can add a PR to make the buildsystems builds run tests by default if you have time. :)

File: packages/gtkmm4.rb [hunk 5]
```diff
@@ -3,23 +3,34 @@
...
+  depends_on 'cairo' # R
+  depends_on 'gcc_lib' # R
+  depends_on 'gdk_pixbuf' # R
+  depends_on 'glibc' # R
+  depends_on 'glibmm_2_68' # R
+  depends_on 'glib' # R
+  depends_on 'gobject_introspection' => :build
+  depends_on 'graphene' # R
+  depends_on 'gtk4' => :build
+  depends_on 'gtk4' # R
```
Comment (Zopolis4) [inline packages/gtkmm4.rb:31]: `gtk4` is listed twice.

**Review by Zopolis4 — DISMISSED** [review]

**Comment by supechicken** [comment]
> Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.

I agree with @satmandu on this point. Especially when most web documents have a nicer format compared with the `man` one

File: packages/gtkmm4.rb (hunk 5 above)
Comment (satmandu) [reply packages/gtkmm4.rb:31]: Good catch! The new build system automatically adds the runtime dep if missing! Hmm. Maybe I will write code at some point to also switch the build dep to runtime dep if it notices the build dep is already there.

**Comment by uberhacker** [comment]
> > Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.
> 
> I agree with @satmandu on this point. Especially when most web documents have a nicer format compared with the `man` one

Not to mention the savings on valuable disk space.

File: packages/atkmm.rb [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
   source_url 'https://gitlab.gnome.org/GNOME/atkmm.git'
   git_hashtag version
   binary_compression 'tar.zst'
 
   binary_sha256({
-    aarch64: '049d1efb8c411ac36328983605a54d54e3bc45587aa3ac9157f4b629be75f720',
-     armv7l: '049d1efb8c411ac36328983605a54d54e3bc45587aa3ac9157f4b629be75f720',
+    aarch64: '9224b82b6db207ebcb652edeb5403eec1be918f7b395d449c2b4a6bd04298a37',
+     armv7l: '9224b82b6db207ebcb652edeb5403eec1be918f7b395d449c2b4a6bd04298a37',
        i686: '4ceaac05ebc9ae13009b10103249356131e260b7ce8ed3a3e57ad91f73f174e2',
```
Comment (uberhacker) [inline packages/atkmm.rb:16]: Missing i686 update?

Comment (satmandu) [reply packages/atkmm.rb]: i686 removed... it doesn't make sense to do gnome updates for i686...

File: packages/gtkmm4.rb [hunk 7]
```diff
@@ -3,23 +3,31 @@
...
+  depends_on 'gdk_pixbuf' # R
+  depends_on 'glibc' # R
+  depends_on 'glibmm_2_68' # R
+  depends_on 'glib' # R
+  depends_on 'gobject_introspection' => :build
+  depends_on 'graphene' # R
+  depends_on 'gtk4' # R
+  depends_on 'libsigcplusplus3' # R
+  depends_on 'pangomm_2_48' # R
+  depends_on 'vulkan_headers' => :build
```
Comment (uberhacker) [inline packages/gtkmm4.rb:32]: Disable documentation?  Filelists are huge.

File: packages/atkmm.rb (hunk 6 above)
Comment (uberhacker) [reply packages/atkmm.rb]: I see it was deleted.  Great!

**Comment by uberhacker** [comment]
> > > Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.
> > 
> > 
> > I agree with @satmandu on this point. Especially when most web documents have a nicer format compared with the `man` one
> 
> Not to mention the savings on valuable disk space.

It might be helpful to provide a link to the online documentation in postinstall?

**Comment by satmandu** [comment]
> > > > Documentation is available on the web for libraries. We don't need to install it locally. The goal is to minimize size for library packages.
> > > 
> > > 
> > > I agree with @satmandu on this point. Especially when most web documents have a nicer format compared with the `man` one
> > 
> > Not to mention the savings on valuable disk space.
> 
> It might be helpful to provide a link to the online documentation in postinstall?

It's elementary to Google documentation on a library...

**Comment by Zopolis4** [comment]
:woman_shrugging: the room is against me on this one, let's disable documentation.

File: packages/msgpack_c.rb [hunk 8]
```diff
@@ -1,50 +1,38 @@
...
-require 'package'
+require 'buildsystems/cmake'
 
-class Msgpack_c < Package
+class Msgpack_c < CMake
   description 'An efficient object serialization library'
   homepage 'https://msgpack.org/'
-  version '4.0.0'
+  version '6.0.0'
   license 'custom:Boost'
```
Comment (Zopolis4) [inline packages/msgpack_c.rb:10]: From what I can see on the repo, its just the version 1.0 of the Boost Software License (BSL-1.0), nothing custom.

File: packages/gtkmm4.rb (hunk 7 above)
Comment (satmandu) [reply packages/gtkmm4.rb:32]: Done.

**Review by uberhacker — APPROVED** [review]

**Review by uberhacker — APPROVED** [review]

**Pull request merged**